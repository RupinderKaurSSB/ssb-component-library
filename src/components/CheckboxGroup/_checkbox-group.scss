.ssb-checkbox-group {
  @include roboto;
  display: inline-block;

  .boxes {
    display: flex;
  }

  .checkbox-group-header{
	font-size: 16px;
	font-weight: bold;
	margin-bottom: 8px;
  }
}
