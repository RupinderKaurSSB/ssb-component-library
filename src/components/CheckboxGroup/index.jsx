import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Checkbox from '../Checkbox/index.jsx';
import Title from '../Title/index.jsx';
import {Â roboto } from '../../style/mixins';

const CheckboxGroup = ({
	header, items, onChange, selectedValues,
}) => {
	const [selected, updateSelected] = useState(selectedValues);

	useEffect(() => {
		onChange(selected);
	}, [selected]);

	const setSelected = sel => {
		const newArr = [...selected];
		if (selected.includes(sel)) {
			newArr.splice(selected.indexOf(sel), 1);
		} else {
			newArr.push(sel);
		}
		updateSelected(newArr);
	};

	return (
		<CheckboxGroupWrapper>
			{header && <Title size={5}>{header}</Title>}
			{items.map((it, index) => (
				<Checkbox
					key={it.value}
					index={index}
					selected={selected.includes(it.value)}
					value={it.value}
					callback={setSelected}
					disabled={it.disabled}
				>{it.label}
				</Checkbox>
			))}
		</CheckboxGroupWrapper>
	);
};

const CheckboxGroupWrapper = styled.div`
	${roboto};
	display: inline-block;

	fieldset {
	  border: 0;
	}
  
	h5 {
	  margin-bottom: 10px;
	  padding-left: 10px;
	}
  
`;

CheckboxGroup.defaultProps = {
	onChange: () => {},
	selectedValues: [''],
};

CheckboxGroup.propTypes = {
	header: PropTypes.string,
	items: PropTypes.arrayOf(PropTypes.shape({
		label: PropTypes.string,
		value: PropTypes.string,
	})).isRequired,
	onChange: PropTypes.func,
	selectedValues: PropTypes.array,
};

export default CheckboxGroup;
