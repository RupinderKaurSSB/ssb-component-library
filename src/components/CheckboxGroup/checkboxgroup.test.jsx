import React from 'react';
import {shallow} from 'enzyme';
import CheckboxGroup from './index';

const items = [
	{
		label: 'Item 1',
		value: 'item1',
	}, {
		label: 'Item 2',
		value: 'item2',
	}, {
		label: 'Item 3',
		value: 'item3',
		disabled: true,
	},
];

describe('CheckboxGroup component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<CheckboxGroup header="Header" items={items} />);
		expect(wrapper).toMatchSnapshot();
	});
});