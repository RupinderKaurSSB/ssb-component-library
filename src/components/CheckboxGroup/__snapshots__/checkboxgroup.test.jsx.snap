// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup component Matches the snapshot 1`] = `
<div
  className="ssb-checkbox-group"
>
  <Title
    className="checkbox-group-header no-margin"
    negative={false}
    size={5}
  >
    Header
  </Title>
  <div
    className="boxes flex-column"
  >
    <Checkbox
      callback={[Function]}
      index={0}
      key="item1"
      selected={false}
      tabIndex={0}
      value="item1"
    >
      Item 1
       
      0
    </Checkbox>
    <Checkbox
      callback={[Function]}
      index={1}
      key="item2"
      selected={false}
      tabIndex={0}
      value="item2"
    >
      Item 2
       
      1
    </Checkbox>
    <Checkbox
      callback={[Function]}
      disabled={true}
      index={2}
      key="item3"
      selected={false}
      tabIndex={0}
      value="item3"
    >
      Item 3
       
      2
    </Checkbox>
  </div>
</div>
`;
